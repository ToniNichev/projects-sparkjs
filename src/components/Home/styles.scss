.title {
  background: rgb(141, 141, 172);
  color: white;
  text-align: center;
  font-family: MyFont;
}

.wrapper {
  background: blue;
  display:flex;
}

.leftRail {
  width:60%;
  border: 1px solid blue;
}

.rightRail {

}