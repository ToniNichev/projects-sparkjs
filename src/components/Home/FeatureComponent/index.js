import React from 'react';
import styles from './styles.scss';


export default (
  <div className={styles.wrapper}>
    Feature component is now enabled
  </div>   
)
