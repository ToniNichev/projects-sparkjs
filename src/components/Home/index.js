import React from 'react';
import styles from './styles.scss';
import MockAPI from './__mocks__/api-data.json';


const Renderer = ({title}) => {
  const namesArray = typeof global.__API_DATA__ !== 'undefined' ? global.__API_DATA__.home : MockAPI.home;
  const names = namesArray.map( (data) => { return data.first_name }).join(',');
  return (
    <div className={styles.wrapper}>
        <div className={styles.leftRail}>
          <div className={styles.title}>{title}</div>
            <p>{names}</p>
        </div>
        <div className={styles.rightRail}>
          <p>right rail</p>
        </div>

    </div>
  );
}

export default Renderer;