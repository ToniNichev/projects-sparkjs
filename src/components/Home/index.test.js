import React from 'react';
import { shallow } from 'enzyme';
import Home from './index';
import toJson from 'enzyme-to-json';
describe('Testing Home component', () => {
    it('renders as expected!', () => {
      const wrapper = shallow(
        <Home title="Title test" />
      );
     expect(toJson(wrapper)).toMatchSnapshot();
   });
});