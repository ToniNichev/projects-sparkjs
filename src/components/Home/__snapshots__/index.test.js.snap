// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home component renders as expected! 1`] = `
<div>
  <div>
    <div>
      Title test
    </div>
    <p>
      Jhon
    </p>
  </div>
  <div>
    <p>
      right rail
    </p>
  </div>
</div>
`;
